source/dbo/Functions/Reusable Notification/ufnGetServiceResponsible.sql

```diff
@@ -0,0 +1,37 @@
+﻿CREATE FUNCTION [dbo].[ufnGetServiceResponsible]
+(
+	@TimePeriod int,
+	@NotificationId int,
+	@StartDate DATETIME,
+	@DefaultService NVARCHAR(150)
+)
+RETURNS NVARCHAR(150)
+AS
+BEGIN
+	DECLARE @ReturnValue NVARCHAR(150) = NULL
+
+	SET @ReturnValue = 
+		COALESCE (
+			(
+			SELECT TOP 1 trans.TbServiceName
+			FROM AllTransfers trans
+			WHERE trans.NotificationId = @NotificationId
+				AND trans.EventDate = DATEADD(DAY, -1, DATEADD(YEAR, @TimePeriod, @StartDate))
+				AND trans.TransferType = 'TransferIn'
+			ORDER BY EventDate DESC
+			),
+			(
+			SELECT TOP 1 trans.TbServiceName
+			FROM AllTransfers trans
+			WHERE trans.NotificationId = @NotificationId
+				AND trans.EventDate >= DATEADD(DAY, -1, DATEADD(YEAR, @TimePeriod, @StartDate))
+				AND trans.TransferType = 'TransferOut'
+			ORDER BY EventDate
+			),
```

Comment:
I don't think we need these two cases, I think it's equivalent to:
```
SELECT TOP 1 trans.TbServiceName
			FROM AllTransfers trans
			WHERE trans.NotificationId = @NotificationId
				AND trans.EventDate > DATEADD(DAY, -1, DATEADD(YEAR, @TimePeriod, @StartDate))
				AND trans.TransferType = 'TransferOut'
			ORDER BY EventDate
```
i.e. if there's a transfer on the day then the transfer in should match the transfer out of the next transfer (and if there isn't another transfer then it should match the current TB service)

Reply:
Having thought about it a little more, they won't be equivalent if the transfers on the record aren't consistent with the current TB service, but in that case I think it's better to prefer the current TB service anyway.